utils/imagenet_topk_accuracy_driver.py

```diff
@@ -0,0 +1,318 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import pexpect
+import shutil
+import sys
+import tempfile
+
+import PIL.Image as Image
+
+import torchvision.transforms as transforms
+
+parser = argparse.ArgumentParser(description='Glow image-classifier Driver for TopK ImageNet Calculation')
+
+parser.add_argument('--validation-images-dir', metavar='DIR', required=True,
+                    help=" Path to the directory containing the validation set "
+                    "of images. Subdirectories are expected to be organized "
+                    "such that when sorted their index corresponds to their "
+                    "label. For example, if the validation_images_dir contains "
+                    "{'abc/', 'def/', 'ghi/'}, then this should correspond to "
+                    "labels {0, 1, 2} respectively.")
+
+parser.add_argument('--batch-size', default=1, type=int, metavar='N',
+                    help="Batch size for use with the model. The total number "
+                    "of images in the validation_images_dir should be "
+                    "divisible by the batch size.")
+
+parser.add_argument('--only-resize-and-save', default=False,
+                    action='store_true', help="Use to pre-process images "
+                    "to 224x224. Saves the images to "
+                    "the validation_images_dir/processed/")
+
+parser.add_argument('--resize-input-images', default=False,
+                    action='store_true', help="Resize and center-crop images "
+                    "at runtime to 224x224.")
+
+parser.add_argument('--verbose', default=False,
+                    action='store_true', help="Verbose printing.")
+
+parser.add_argument('--image-classifier-cmd', default="",
+                    help="Command to use for running the image-classifier, "
+                    "including the binary and all of its command lime "
+                    "parameters.")
+
+def pil_loader(path):
+    # open path as file to avoid ResourceWarning
+    # (https://github.com/python-pillow/Pillow/issues/835)
+    with open(path, 'rb') as f:
+        img = Image.open(f)
+        return img.convert('RGB')
+
+def accimage_loader(path):
+    import accimage
+    try:
+        return accimage.Image(path)
+    except IOError:
+        # Potentially a decoding problem, fall back to PIL.Image
+        return pil_loader(path)
+
+def default_image_loader(path):
+    from torchvision import get_image_backend
```

Comment:
please don't do import-not-at-top

Reply:
I removed this one, however the accimage above is still there -- it seems like a conditional import based on the if below (I took this code snippet from a different run script). I'm not sure of a good way to get around this, do you have any suggestion?

Reply:
For accimage it's actually fine: it may not be present.